stdlib/public/SwiftRemoteMirror/CMakeLists.txt

```diff
@@ -1,15 +1,29 @@
 # libswiftRemoteMirror.dylib should not have runtime dependencies; it's
 # always built as a shared library.
 if(SWIFT_BUILD_DYNAMIC_STDLIB OR SWIFT_BUILD_REMOTE_MIRROR)
-  add_swift_target_library(swiftRemoteMirror
-                    SHARED TARGET_LIBRARY DONT_EMBED_BITCODE NOSWIFTRT
-                    SwiftRemoteMirror.cpp
-                    LINK_LIBRARIES
-                      swiftReflection
-                    C_COMPILE_FLAGS
-                      ${SWIFT_RUNTIME_CXX_FLAGS} -DswiftRemoteMirror_EXPORTS
-                    LINK_FLAGS
-                      ${SWIFT_RUNTIME_LINK_FLAGS}
-                    INSTALL_IN_COMPONENT
-                      swift-remote-mirror)
+  if (SWIFT_BUILD_STDLIB)
+    add_swift_target_library(swiftRemoteMirror
+                      SHARED TARGET_LIBRARY DONT_EMBED_BITCODE NOSWIFTRT
+                      SwiftRemoteMirror.cpp
+                      LINK_LIBRARIES
+                        swiftReflection
+                      C_COMPILE_FLAGS
+                        ${SWIFT_RUNTIME_CXX_FLAGS} -DswiftRemoteMirror_EXPORTS
+                      LINK_FLAGS
+                        ${SWIFT_RUNTIME_LINK_FLAGS}
+                      INSTALL_IN_COMPONENT
+                        swift-remote-mirror)
+  else()
+    add_swift_host_library(swiftRemoteMirror
+                      STATIC
+                      SwiftRemoteMirror.cpp
+                      LINK_LIBRARIES
+                        swiftReflection
+                      C_COMPILE_FLAGS
+                        ${SWIFT_RUNTIME_CXX_FLAGS} -DswiftRemoteMirror_EXPORTS
+                      LINK_FLAGS
+                        ${SWIFT_RUNTIME_LINK_FLAGS}
+                      INSTALL_IN_COMPONENT
+                        swift-remote-mirror)
+  endif()
```

Comment:
This doesn't really make sense to me.  Did you try building with the standard library disabled?  I think that rather than doing this, we should actually just add a secondary build of swift-remote-mirror as a static library for the host.  This should not get installed at all and always use that in the lldb build.  This would also simplify the lldb build (much like `swiftReflection`)

Reply:
@compnerd +1!

Reply:
Of course I tried it. That's how we deliver LLDB internally (we don't need to build the standard library as it's provided by the build of the compiler). I don't care that it gets installed, I don't think that's needed. The code change I made here was to adapt to your refactoring in swiftReflection, where there won't be a target library if you don't build the library but there will be a host library. The logic here needs to mirror what's in swiftReflection, otherwise it won't link.

I'm worried that adding the host library unconditionally will end up with duplicated symbols, but I can try that (modifying swiftReflection in the same way, making its host build unconditional.

Reply:
Sorry, I wasn't trying to be flippant or disrespectful, I honestly did not expect `SWIFT_BUILD_STDLIB` to be defined in side the conditional.  But, very cool, glad to know that the work related to that is useful to you :-).  Yeah, I agree that the installation isn't needed.  I'm suggesting that we model this even closer to that change in that we build a host version (note that the name will not have a suffix there).  It shouldn't end up with any duplicated symbols, the host build would not be used by the compiler or the stdlib, but will be consumed by lldb (the minor cost of building it is really just noise).

Reply:
Put up #21904 which should allow you to test against lldb pretty easily I hope.

Reply:
I didn't mean to imply you were disrespectful :-) It was merely a "do you really think I would be touching this if I hadn't seen it fail?". I'm trying out your other PR.